pkg/asset/installconfig/basedomain.go

```diff
@@ -28,17 +36,28 @@ func (a *baseDomain) Dependencies() []asset.Asset {
 func (a *baseDomain) Generate(parents asset.Parents) error {
 	platform := &platform{}
 	parents.Get(platform)
-
-	if platform.AWS != nil {
```

Comment:
this is changing existing behavior of the function. please only add the special handling for azure without introducing any other behavior.

Reply:
I changed this because every other asset follows this pattern for platform specific branch.
I'll revert to a `if platform != nil` chain if that's better.

Reply:
it's not actually changing behavior except for the unknown platform type error, which seemed legitimate.

Reply:
> it's not actually changing behavior except for the unknown platform type error, which seemed legitimate.

but that's still a behavior change. i would like to keep them constant as much as we can.

> I changed this because every other asset follows this pattern for platform specific branch.
> I'll revert to a if platform != nil chain if that's better.

it's fine to switch to `platform.CurrentName()` switch but don't change other behavior of this function.

Reply:
so, no error and we are OK? please be specific in your review, it is hard to guess what you expect.

Reply:
> so, no error and we are OK? please be specific in your review, it is hard to guess what you expect.

> > it's not actually changing behavior except for the unknown platform type error, which seemed legitimate.
> 
> but that's still a behavior change. i would like to keep them constant as much as we can.

Hmm not sure how much more **specific** i need to be than above ^^

drop the switch cases with platforms that don't have special casing for basedomain and the error on default.
you change of using `platform.CurrentName` to switch on aws and azure is fine.

> > I changed this because every other asset follows this pattern for platform specific branch.
> > I'll revert to a if platform != nil chain if that's better.
> 
> it's fine to switch to `platform.CurrentName()` switch but don't change other behavior of this function.

Reply:
fixed